api/proto/config/devconfig.proto

```diff
@@ -155,6 +155,12 @@ message EdgeDevConfig {
   // This ensures that device will not accidentally revert back to an older configuration.
   // Does not apply to legacy override.json and usb.json mechanisms.
   google.protobuf.Timestamp config_timestamp = 40;
+
+  // If set, indicates a hostname/IPv4/IPv6 address and optional port number at
+  // which EVE will request for a Local Operator Console (LOC). The format is
+  // similar to the format of the `local_profile_server` field.
+  // If the port number is not specified, it will default to 443.
+  string loc_server = 41;
```

Comment:
What are the members of the structure?

Reply:
> IMO, we should go for proper structure instead of string.

I agree, even if there is only one member for the start. Very likely we will be adding more.

I also wonder what to do about the local profile server. My expectation was that it would be reused and basically extended with more features. I personally would prefer if we could avoid having both LOC and LPS with separate APIs, otherwise it could be confusing. Perhaps we could mark `local_profile_server` and `profile_server_token` as deprecated, stop using the strange "local profile server" title and recommend controller(s) moving to LOC config API?

Lastly (and unrelated to this PR), I think it would be nice if we could provide an open-source variant of LOC under a new lf-edge repo (witha very limited functionality, e.g. only with few CLIs), like Adam is to Zedcloud. Would do you think?

Reply:
@milan-zededa adding a structure is fine, just expected more clarity on what we put there. Personally I dont like to produce extra instances "just in case".

LPS name is very strange and confusing, I agree. The idea of LOC API is to have something similar to what eve and controller have, reuse the API as much as possible. I support the idea to have only one instance, but LPS is already used, so it will be painful to get rid of it. Also not clear how often we do API changes on the LPS side. If nothing was changed since the beginning - then probably nothing to worry about and we just smoothly introduce LOC with all the LPS features.

Instead of having another (open-source) LOC implementation what if we focus on Adam improvements? For example support N nodes for 1 Adam instance? At the end Adam can act as a LOC (the key here is that API should not differ, so supported by eve by default). With Adam we can solve this "config out of sync" problem (how - is out of scope of this discussion), and make Adam work in any combination: eve nodes + commercial controller + Adam (acts as LOC, deployed in local environment) or eve nodes + Adam (acts as a controller, deployed in cloud) + Adam (acts as LOC, deployed in local environment). Then full open source stack. Thoughts?

Reply:
Regarding the simple open-source LOC implementation - I think we can decide this later. Adam is quite simple and would require quite a bit of work to get it to support N nodes and integrate with LOC. The idea behind open-source LOC was more to have some basic template for open-source community, in case someone will want to implement their own.